packages/vuetify/src/components/VDataIterator/VDataIterator.ts

```diff
@@ -218,6 +211,25 @@ export default mixins(
       this.selection = selection
       emit && this.$emit('item-selected', { item, value })
     },
+    multipleSelect (value = true, emit = true, selection: any, index: number): void {
+      if (index < this.lastEntry) {
+        for (let i = index; i <= this.lastEntry; i++) {
+          const currentItem = this.selectableItems[i]
+          const key = getObjectValueByPath(currentItem, this.itemKey)
+          if (value) selection[key] = currentItem
+          else delete selection[key]
+          emit && this.$emit('item-selected', { currentItem, value })
+        }
+      } else if (index > this.lastEntry) {
+        for (let i = this.lastEntry; i <= index; i++) {
+          const currentItem = this.selectableItems[i]
+          const key = getObjectValueByPath(currentItem, this.itemKey)
+          if (value) selection[key] = currentItem
+          else delete selection[key]
+          emit && this.$emit('item-selected', { currentItem, value })
+        }
```

Comment:
Try to refactor this into something more DRY. The only difference between the ifs is the for loop definition